### ISSUE TITLE
about BHM_QMC

### ISSUE THREAD

**Issue by sassan72** [RiceReallyGood/BHM_QMC#1]
this code runs but the results are false. the output for single electron density, winding number are 0 and superfluid stiffness is Nan.

**Comment by regymm** [comment]
Hi, I happened to work on this code recently, and I found the 0 winding number and nan superfluid stiffness are because the system is still thermalizing(default thermalizing step 5000000, so it takes a while to finish). I haven't read the details yet but the results look good so far.

**Comment by sassan72** [comment]
Hi Peter. Good. Could you tell me which parameter (parameters) did you change in main file to see the result? I know the code has been written ver well. Could you please send me the corrected main file to check it? Thanks.

**Comment by regymm** [comment]
```cpp
    directed_worm_algorithm::count_type thermalization_sweeps = 5000000;
    directed_worm_algorithm::count_type measurement_sweeps = 5000000;
```
These two lines in `usewl.cpp` determine the sweep number, reduce them will make the program end quicker so results can be seen.

**Comment by sassan72** [comment]
#include "dwa.hpp"
#include <iostream>
#include <fstream>
#include <cstdlib>
using namespace std;

int main()
{
    double t = 0.31;
    double beta = 1.0 / 0.01;
    double U = 1;
    double mu = 0.4 * U;
    unsigned short dim = 2;
    unsigned int Nl = 20;
    unsigned short max_state = 40;
    double time_of_flight = 0;
    double harmonic_shape = 0.0 * U;
    directed_worm_algorithm::count_type thermalization_sweeps = 500;
    directed_worm_algorithm::count_type measurement_sweeps = 500;
    unsigned short measurement_time_steps = beta / 0.1;
    /* directed_worm_algorithm(double beta_, double t_, double U_, double mu_, unsigned short dim_, unsigned int Nl_,
                            unsigned short max_state_, count_type _thermalization_sweeps, count_type _measurement_sweeps,
                            unsigned short measurement_time_steps_, double time_of_flight_, double harmonic_shape_); */
    directed_worm_algorithm mysystem(beta, t, U, mu, dim, Nl, max_state,
                                     thermalization_sweeps, measurement_sweeps, measurement_time_steps, time_of_flight, harmonic_shape);
    mysystem.print_result();
  

    
}

Actually, I run the code with reduced number of sweeps. But I can still see these results for average density:0 and superfluid stiffness:nan

**Comment by sassan72** [comment]
you're right. thermalization_Sweeps and measurement_sweeps must be reduced. But for example if you consider directed_worm_algorithm::count_type thermalization_sweeps = 50;
directed_worm_algorithm::count_type measurement_sweeps = 50;
which is a small value to run the code. And if you only consider my system.print_result(), you will get average density=0 and superfluid stiffness=nan.

**Comment by PUDongOK** [comment]
Could you please tell me whether this code runs on Windows system or not? What third-party libraries does it rely on or which is the header file and which is the source file? I have just come into contact with C++. I can't run this code on Windows and don't know how to compile it on Linux.

**Comment by PUDongOK** [comment]
Severity Code Description Item File Line Suppressed Status
Error C3861 'setiosflags': identifier Bose-Hubbard not found E:\20230916\C++_learning\worm\Bose-Hubbard\dwa.hpp 549
